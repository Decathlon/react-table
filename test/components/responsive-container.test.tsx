/// <reference path="../typings/tests-entry.d.ts" />
import * as React from "react";
import { createRenderer } from "react-test-renderer/shallow";

import ResponsiveContainer from "../../src/components/responsive-container";

describe("Cell component", () => {
  test("should render the default responsive", () => {
    const props = {
      className: "foo-class-name",
      children: () => <div>Foo</div>
    };
    const shallowRenderer = createRenderer();
    shallowRenderer.render(<ResponsiveContainer {...props} />);
    const rendered = shallowRenderer.getRenderOutput();
    expect(rendered).toMatchSnapshot();
  });
  // "behaviours dependent on rendered element sizes cannot be tested with jest/enzyme/jsdom"
  // "jsdom doesn't support layout. This means measurements like this will always return 0 as it does here"
});
