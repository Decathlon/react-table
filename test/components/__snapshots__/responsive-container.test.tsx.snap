// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell component should render the default responsive 1`] = `
<div
  className="responsive-container foo-class-name"
/>
`;
